src/server/api.rs

```diff
@@ -90,32 +87,43 @@ async fn login_credentials(
 
     login(&r.email, &r.password, &pool)
         .and_then(|account| {
-            let token = account.issue_token(&pool).unwrap();
-            let user_string = serde_json::to_string(&account)
+            account.issue_token(&pool)?;
+            let details = get_account_details(&account.email, &pool).unwrap();
+            let user_string = serde_json::to_string(&details)
                 .map_err(|_| ThothError::InternalError("Serder error".into()))?;
             id.remember(user_string);
-            Ok(HttpResponse::Ok().json(Login(Session { token })))
+            Ok(HttpResponse::Ok().json(details))
         })
         .map_err(error::ErrorUnauthorized)
 }
 
 #[post("/account/token/renew")]
 async fn login_session(
-    payload: web::Json<LoginSession>,
     token: DecodedToken,
     id: Identity,
     pool: web::Data<PgPool>,
 ) -> Result<HttpResponse, Error> {
-    let r = payload.into_inner();
-    token.jwt.as_ref().ok_or(ThothError::Unauthorised)?;
+    let email = match id.identity() {
```

Comment:
Identical check performed in `account_details` below - could this be commonised?

Reply:
Yes, definitely. I want to have the checked moved to an implementation of the `actix_web::FromRequest` trait down the line

Reply:
I'm happy to tackle this now, though I'm not sure how that'll affect your changes to `Context`

Reply:
I'm not sure either, although they're now published under #188 if that makes it clearer :) Fine to leave it if that's easiest, since changing it's already in our roadmap.

Reply:
Ok, let's try to merge the two PRs and then we'll have a separate one where we tidy up the identity checks